apps/studio/src/lib/db/clients/duckdb.ts

```diff
@@ -0,0 +1,944 @@
+          is_nullable,
+          column_default,
+          comment
+        FROM duckdb_columns
+      `;
+
+    if (table) {
+      params.push({ column: "table_name", value: table });
+    }
+
+    if (schema) {
+      params.push({ column: "schema_name", value: schema });
+    }
+
+    if (params.length > 0) {
+      const conditions = params.map((p) => `${p.column} = ?`).join(" AND ");
+      query += ` WHERE ${conditions}`;
+    }
+
+    const { data } = await this.driverExecuteSingle(query, {
+      params: [...params.map((p) => p.value)],
+    });
+    return data.map(
+      (row: any): ExtendedTableColumn => ({
+        schemaName: row.schema_name,
+        tableName: row.table_name,
+        columnName: row.column_name,
+        ordinalPosition: row.column_index,
+        dataType: row.data_type,
+        nullable: row.is_nullable,
+        defaultValue: row.column_default,
+        hasDefault: !_.isNil(row.column_default),
+        comment: row.comment,
+      })
+    );
+  }
+
+  async listTableTriggers(
+    _table: string,
+    _schema: string
+  ): Promise<TableTrigger[]> {
+    // DuckDB doesn't support table triggers
+    // https://github.com/duckdb/duckdb/issues/750
+    return [];
+  }
+
+  async listTableIndexes(
+    table: string,
+    schema?: string
+  ): Promise<TableIndex[]> {
+    const { data } = await this.driverExecuteSingle(
+      `
+        SELECT
+          schema_name,
+          table_name,
+          index_name,
+          index_oid,
+          comment,
+          is_unique,
+          is_primary
+        FROM duckdb_indexes
+        WHERE table_name = ?
+          AND schema_name = ?
+      `,
+      { params: [table, schema] }
+    );
+
+    return data.map((row) => ({
+      id: row.index_oid,
+      table: row.table_name,
+      schema: row.schema_name,
+      name: row.index_name,
+      columns: [], // TODO
+      unique: row.is_unique,
+      primary: row.is_primary,
+    }));
+  }
+
+  async listSchemas(filter?: SchemaFilterOptions): Promise<string[]> {
+    const filterQuery = buildSchemaFilter(filter);
+    const { data } = await this.driverExecuteSingle(`
+      SELECT DISTINCT schema_name
+      FROM information_schema.schemata
+      ${filterQuery ? "WHERE " + filterQuery : ""}
+    `);
+    return data.map((row) => row.schema_name);
+  }
+
+  async getTableReferences(_table: string, _schema: string): Promise<string[]> {
+    return []; // TODO: not implemented yet
+  }
+
+  // TODO check this. Since duckdb is a OLAP database, it might not have these kinda things (foreign keys).
+  async getTableKeys(_table: string, _schema: string): Promise<TableKey[]> {
+    // toTable: string;
+    // toSchema: string;
+    // toColumn: string;
+    // fromTable: string;
+    // fromSchema: string;
+    // fromColumn: string;
+    // constraintName?: string;
+    // onUpdate?: string;
+    // onDelete?: string;
+    // const { data } = await this.driverExecuteSingle(``);
+    return [];
+  }
```

Comment:
Did you check this? If it's not supported I'd just remove all the commented stuff and add a note.

Reply:
Turns out it supports foreign keys lol.